### PR TITLE
17572/17565 - SubProduct Application Status Messaging

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:*
https://github.com/bcgov/entity/issues/17572
https://github.com/bcgov/entity/issues/17565

*Description of changes:*
* Applies SubProduct status messaging for Pending, Approved and Rejected mhr sub product applications
* Includes new common component for notification/caution/alert messages
* Unit tests
*Note:* - No feature flag required for this content - protected behind ability to submit applications.

Screens for context:

Screenshot 2024-01-26 at 7 42 34 AM
Screenshot 2024-01-26 at 7 42 06 AM
Screenshot 2024-01-26 at 7 39 12 AM
Screenshot 2024-01-26 at 7 35 22 AM
Screenshot 2024-01-26 at 7 35 10 AM
Screenshot 2024-01-26 at 7 34 53 AM
Screenshot 2024-01-26 at 7 34 40 AM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/account-settings/product/ProductPackage.vue [hunk 1]
```diff
@@ -115,19 +137,49 @@
...
+import CautionBox from '@/components/auth/common/CautionBox.vue'
 import { KCUserProfile } from 'sbc-common-components/src/models/KCUserProfile'
 import ModalDialog from '@/components/auth/common/ModalDialog.vue'
 import Product from '@/components/auth/common/Product.vue'
+import { ProductStatusMsgContentIF } from '@/models/external'
 import { useOrgStore } from '@/stores/org'
 import { useUserStore } from '@/stores/user'
+import { userAccessDisplayNames } from '@/resources/QualifiedSupplierAccessResource'
 
 @Component({
```
Comment (seeker25) [inline auth-web/src/components/auth/account-settings/product/ProductPackage.vue:169]: if you have the spare cycles, comp-api, if not no worries

**Review by seeker25 — APPROVED** [review]

File: auth-web/src/components/auth/account-settings/product/ProductPackage.vue (hunk 1 above)
Comment (cameron-eyds) [reply auth-web/src/components/auth/account-settings/product/ProductPackage.vue:169]: I can do this!

File: auth-web/src/components/auth/account-settings/product/ProductPackage.vue [hunk 2]
```diff
@@ -115,183 +137,261 @@
...
-  public expandedProductCode: string = ''
-  public AccountEnum = Account
-  public orgProductsFees:any = ''
-  public orgProductFeeCodes:any = ''
-  public isProductActionCompleted: boolean = false
+    const {
+      currentOrganization,
+      productList,
+      currentSelectedProducts
+    } = storeToRefs(useOrgStore())
```
Comment (cameron-eyds) [inline auth-web/src/components/auth/account-settings/product/ProductPackage.vue:191]: A way of preserving reactivity between the composable properties and it's local use here, without redefining each as a local computed value.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2694-k06q640d.web.app

**Review by doug-lovett — APPROVED** [review]
Looks great.

**Review by dimak1 — APPROVED** [review]

**Pull request merged**